lib/dispatch.go

```diff
@@ -71,7 +79,10 @@ func (isw *InstanceSourceWrap) Dispatch(ctx context.Context, method string, para
 
 func (inst *Instance) dispatchMethodCall(ctx context.Context, method string, param interface{}, source string) (res interface{}, cur Cursor, err error) {
 	if inst == nil {
-		return nil, nil, fmt.Errorf("instance is nil, cannot dispatch")
+		return nil, nil, ErrDispatchNilInstance
+	}
+	if param == nil || (reflect.ValueOf(param).Kind() == reflect.Ptr && reflect.ValueOf(param).IsNil()) {
+		return nil, nil, ErrDispatchNilParam
```

Comment:
Here is the MEGA fun part, detailed in the commit message and https://github.com/qri-io/qri/pull/1736#issue-608221386

Reply:
That is SUPER interesting. Great catch, so subtle. 👏